src/tree-controller.ts

```diff
@@ -0,0 +1,51 @@
+import { TreeService } from './tree.service';
+import { Tree } from './tree';
+import { TreeModel } from './tree.types';
+import { NodeEditableEvent, NodeEditableEventAction } from './editable/editable.events';
+import { TreeInternalComponent } from './tree-internal.component';
+import { Subject, Observable } from 'rxjs/Rx';
+import { Injectable, Inject, ElementRef } from '@angular/core';
+import * as _ from 'lodash';
+
+export class TreeController {
+  constructor (private treeInternalComponent: TreeInternalComponent) { }
+
+  public select(e: MouseEvent = new MouseEvent('click')): void {
+    if (this.treeInternalComponent && !this.treeInternalComponent.isSelected && typeof this.treeInternalComponent.onNodeSelected === 'function') {
+      this.treeInternalComponent.onNodeSelected(e);
+    }
+  }
+
+  public switchFoldingType(): void {
+    if (this.treeInternalComponent && typeof this.treeInternalComponent.onSwitchFoldingType === 'function') {
+      this.treeInternalComponent.onSwitchFoldingType();
+    }
+  }
+
+  public expand(): void {
+    if (this.treeInternalComponent && this.treeInternalComponent.tree) {
```

Comment:
@lovery It will be easier to put this into constructor 
```

constructor (private treeInternalComponent: TreeInternalComponent) {
  this.tree = this.treeInternalComponent.tree
}

```